docs/user-guide/configuring.md

```diff
@@ -372,6 +372,36 @@ your-project
   └── test.js
 ```
 
+By default, ESLint will look for configuration files in all parent folders up to the home directory. This can be useful if you want all of your projects to follow a certain convention, but can sometimes lead to unexpected results. To limit ESLint to a specific project, place `"root": true` inside the `eslintConfig` field of the `package.json` file or in the `.eslintrc` file at your project's root level.
```

Comment:
Nit: ESLint searches all the way up to the root directory, then only looks at `~/.eslintrc` if it hasn't found anything yet, though here you probably only want to mention that it goes up to the root directory.